modules/actions/snapshots/lib/compute-daily-values.ts

```diff
@@ -0,0 +1,31 @@
+import { Prisma } from '@prisma/client';
+import _ from 'lodash';
+
+export const computeDailyValues = (
+    snapshots: Prisma.PrismaPoolSnapshotUncheckedCreateInput[],
+): Prisma.PrismaPoolSnapshotUncheckedCreateInput[] => {
+    // Group snapshots by poolId
+    const groupedByPoolId = _.groupBy(snapshots, 'poolId');
+
+    // Iterate through each group to compute daily values
+    const updatedSnapshots = _.flatMap(groupedByPoolId, (snapshots) => {
+        // Sort snapshots by timestamp for the pool
+        const sortedSnapshots = _.sortBy(snapshots, 'timestamp');
+
+        return sortedSnapshots.map((snapshot, index) => {
+            const previousSnapshot = sortedSnapshots[index - 1];
+
+            // Calculate daily values as the difference from the previous snapshot
+            const volume24h = Math.max((snapshot.totalSwapVolume || 0) - (previousSnapshot?.totalSwapVolume || 0), 0);
```

Comment:
I dont think this works as intended. If we have two snapshots for a pool, one 'partial' from today and one 'finished' from yesterday. You set the volume24hrs of the 'finished' one to `totalSwapVolume`. Or am I missing something?